osutil/unlink.go

```diff
@@ -0,0 +1,61 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package osutil
+
+import (
+	"os"
+	"syscall"
+
+	"github.com/snapcore/snapd/osutil/sys"
+)
+
+// UnlinkMany removes multiple files from a single directory.
+//
+// If dirname is not a directory, this will fail.
+//
+// This will abort at the first removal error (but ENOENT is ignored).
+//
+// Filenames must refer to files. They don't necessarily have to be
```

Comment:
Hm, so I could do: osutil.UnlinkMany("/etc/", ["/var/lib/foo"]) ? If so thats a slightly odd interface, I mean: it seems to me that either it should enforce that all removes are in this basedir (so no path in the filenames) or we don't need a basedir. Or am I missing something?

Reply:
yeah, it's weird, but that's `unlinkat` interface. As any checks we did would be racy, and the function is a convenience thing more than anything else, I didn't think it was worth it

Reply:
I don't strongly mind either way, but something like: `for _, name := range filesnames { if filepath.Base() != name { return fmt.Errorf("can only use base names") }` shouldn't be racy. Or am I misunderstanding the race?

Reply:
Yes, I could do that. In fact I was going to use strings.IndexByte to much the same effect. But then I thought: what does this gain us? I didn't have an answer, so I didn't do it. I still don't have an answer :-) but if there is something to be gained, I'd be glad to add the check.

Reply:
(i missed your comment before merging, fwiw, but i'll gladly push a PR that fixes this if wanted)